src/server/queue_recov_db.c

```diff
@@ -133,18 +119,18 @@ svr_to_db_que(pbs_queue *pque, pbs_db_que_info_t *pdbque, int updatetype)
  *@return !=0    Failure
  */
 static int
-db_to_svr_que(pbs_queue *pque, pbs_db_que_info_t *pdbque)
+db_2_que(pbs_queue *pque, pbs_db_que_info_t *pdbque)
 {
-	pque->qu_qs.qu_name[sizeof(pque->qu_qs.qu_name) - 1] = '\0';
-	strncpy(pque->qu_qs.qu_name, pdbque->qu_name, sizeof(pque->qu_qs.qu_name));
+	strcpy(pque->qu_qs.qu_name, pdbque->qu_name);
 	pque->qu_qs.qu_type = pdbque->qu_type;
-	pque->qu_qs.qu_ctime = pdbque->qu_ctime;
-	pque->qu_qs.qu_mtime = pdbque->qu_mtime;
 
-	if ((decode_attr_db(pque, &pdbque->attr_list, que_attr_def,
-		pque->qu_attr, (int) QA_ATR_LAST, 0)) != 0)
+	if ((decode_attr_db(pque, &pdbque->cache_attr_list, &pdbque->db_attr_list, que_attr_def, pque->qu_attr, (int) QA_ATR_LAST, 0)) != 0)
 		return -1;
 
+	obj_qs_modified(&pque->qu_qs, sizeof(pque->qu_qs), pque->qs_hash);
```

Comment:
Can we collect the return value of the function obj_qs_modified and handle accordingly. It is because this function is also returning -1 in case of failure.

Reply:
Actually, obj_qs_modified does not return error. SHA1 should not fail since we provide it with exact buffer we need, and it merely hashes it. If it fails, there is probably severe problems in the system setup itself, so we simply quit PBS